crates/ruff_python_parser/src/parser/statement.rs

```diff
@@ -868,28 +898,25 @@ impl<'src> Parser<'src> {
 
         helpers::set_expr_ctx(&mut target.expr, ExprContext::Store);
 
-        let simple = target.expr.is_name_expr() && !target.is_parenthesized;
+        let simple = target.is_name_expr() && !target.is_parenthesized;
 
-        // Annotation is actually just an `expression` but we use `expressions`
-        // for better error recovery in case tuple isn't parenthesized.
-        let annotation = self.parse_expression_list();
-
-        if matches!(
-            annotation.expr,
-            Expr::Tuple(ast::ExprTuple {
-                parenthesized: false,
-                ..
-            })
-        ) {
-            self.add_error(
-                ParseErrorType::OtherError("annotation cannot be unparenthesized".into()),
-                annotation.range(),
-            );
-        }
+        let annotation = self.parse_conditional_expression_or_higher();
 
-        let value = self
-            .eat(TokenKind::Equal)
-            .then(|| Box::new(self.parse_expression_list().expr));
+        let value = if self.eat(TokenKind::Equal) {
+            if self.at_expr() {
+                Some(Box::new(self.parse_expression_list().expr))
+            } else {
+                // test_err ann_assign_stmt_missing_rhs
+                // x: int =
+                self.add_error(
+                    ParseErrorType::OtherError("Expected an expression".to_string()),
+                    self.current_token_range(),
+                );
+                None
+            }
+        } else {
+            None
+        };
```

Comment:
I'm adopting this pattern wherever I can. This is to avoid adding an invalid expression node in the AST where `None` is feasible.